src/Components/ImagesTable/Target.tsx

```diff
@@ -0,0 +1,54 @@
+import {
+  useGetClonesQuery,
+  useGetComposeStatusQuery,
+} from '../../store/apiSlice';
+
+type TargetProps = {
+  composeId: string;
+};
+
+const targetOptions = {
+  aws: 'Amazon Web Services',
+  'aws.s3': 'Amazon Web Services',
+  azure: 'Microsoft Azure',
+  gcp: 'Google Cloud Platform',
+  vsphere: 'VMWare',
+  'guest-image': 'Virtualization - Guest image',
+  'image-installer': 'Bare metal - Installer',
+  'edge-commit': 'edge-commit',
+  'edge-installer': 'edge-installer',
+  ami: 'ami',
+  'rhel-edge-commit': 'rhel-edge-commit',
+  'rhel-edge-installer': 'rhel-edge-installer',
+  vhd: 'vhd',
+};
+
+const Target = ({ composeId }: TargetProps) => {
+  const { data: clones, isSuccess: isSuccessClones } =
+    useGetClonesQuery(composeId);
+  const { data: composeStatus, isSuccess: isSuccessTarget } =
+    useGetComposeStatusQuery(composeId);
+  const target = composeStatus?.request?.image_requests[0];
+
+  let tar;
+
+  if (isSuccessTarget) {
+    if (target?.upload_request?.type === 'aws.s3') {
+      tar = targetOptions[target?.image_type];
+    } else if (target?.upload_request?.type === 'aws') {
+      if (isSuccessClones) {
+        tar =
+          targetOptions[target?.upload_request?.type] +
+          ` (${clones.meta.count !== 0 ? clones.meta.count + 1 : 1})`;
+      }
+    } else {
+      tar = targetOptions[target?.upload_request?.type];
+    }
```

Comment:
I think we can actually simplify this by only specialcasing the `aws` and `ami` image_types. If it's `aws` or `ami`, check for clones. Otherwise just use the `image_type` in the image request.

I'd try to drop any use of `upload_request.type` if possible.

Reply:
ami doesnt exist in the schema  : UploadTypes: "aws" | "gcp" | "azure" | "aws.s3";
do I need to add it?